package.json

```diff
@@ -32,9 +35,6 @@
     "typescript": "^2.1.6"
   },
   "dependencies": {
-    "@types/bluebird": "^3.0.37",
-    "@types/lodash": "^4.14.52",
-    "@types/node": "^7.0.7",
```

Comment:
Yes, it's not perfect. On balance the extra weight of a couple of types is pretty tiny though, and much better than having to manually manage all the required `@types` dependencies downstream to get anything to work, imo.

That would be a bit painful if you were using one module like this, but when we have 10s or 100s of TypeScript modules, it's pretty impractical to manually duplicate all their devDependencies types as separate devDependencies at every step downstream. It then gets entirely impossible to manage if anybody requiring anything other than the latest version of each types. If we have a submodule that wants the types for lodash < 4, in a project that wants 4, you _must_ have the `@types/lodash@3` dependency in that module's non-dev dependencies, or there's no way you're going to be able to use it with TS at all.

If we're worried about extra final container size, it's easy to add a deploy step that deletes `node_modules/@types`, and then you have the best of both worlds.

Alternatively, we go further: add our own `typesDependencies` section to package.json, and use some pre/post npm script to manage them separately? Non-trivial though.

The dev/prod split is just because of where the specific types are used: mocha/chai types are in dev because downstream definitely doesn't want them, and everything else is a normal dependency.